x-pack/test/functional/apps/rollup_job/hybrid_index_pattern.js

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import datemath from '@elastic/datemath';
+import expect from '@kbn/expect';
+import mockRolledUpData, { mockIndices } from './hybrid_index_helper';
+
+export default function ({ getService, getPageObjects }) {
+
+  const es = getService('es');
+  const esArchiver = getService('esArchiver');
+  const retry = getService('retry');
+  const PageObjects = getPageObjects(['common', 'settings']);
+
+  describe('hybrid index pattern', function () {
+    //Since rollups can only be created once with the same name (even if you delete it),
+    //we add the Date.now() to avoid name collision.
+    const rollupJobName = `hybrid-index-pattern-test-rollup-job-` + Date.now();
+    const rollupTargetIndexName = `rollup-target-data`;
+    const regularIndexPrefix = `regular-index`;
+    const rollupSourceIndexPrefix = `rollup-source-data`;
+    const now = new Date();
+    const pastDates = [
+      datemath.parse('now-1d', { forceNow: now }),
+      datemath.parse('now-2d', { forceNow: now }),
+      datemath.parse('now-3d', { forceNow: now }),
+    ];
+
+    it('create hybrid index pattern', async () => {
+      //Create data for rollup job to recognize.
+      //Index past data to be used in the test.
+      await pastDates.map(async (day) => {
+        await es.index(mockIndices(day, rollupSourceIndexPrefix));
+      });
+
+      await retry.waitForWithTimeout('waiting for 3 records to be loaded into elasticsearch.', 10000, async () => {
+        const response = await es.indices.get({
+          index: `${rollupSourceIndexPrefix}*`,
+          allow_no_indices: false
+        });
+        return Object.keys(response).length === 3;
+      });
+
+      await retry.try(async () => {
+        //Create a rollup for kibana to recognize
+        await es.transport.request({
+          path: `/_rollup/job/${rollupJobName}`,
+          method: 'PUT',
+          body: {
+            'index_pattern': `${rollupSourceIndexPrefix}*`,
+            'rollup_index': rollupTargetIndexName,
+            'cron': '*/10 * * * * ?',
+            'groups': {
+              'date_histogram': {
+                'fixed_interval': '1000ms',
+                'field': '@timestamp',
+                'time_zone': 'UTC'
+              }
+            },
+            'timeout': '20s',
+            'page_size': 1000
+          }
+        });
+      });
+
+
+      await pastDates.map(async (day) => {
+        await es.index(mockRolledUpData(rollupJobName, rollupTargetIndexName, day));
+      });
+
+      //Add data for today as live data.
+      const futureDates = [
+        datemath.parse('now', { forceNow: now })
+      ];
+
+      //Index live data to be used in the test.
+      await futureDates.map(async (day) => {
```

Comment:
Nit: Because `futureDates` contains only a single value, we don't need an array or `map` any more.

```js
await es.index(mockIndices(datemath.parse('now', { forceNow: now }), regularIndexPrefix));
```

Reply:
Done.